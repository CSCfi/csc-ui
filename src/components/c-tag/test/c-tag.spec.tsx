import { newSpecPage } from '@stencil/core/testing';
import { CTag } from '../c-tag';

describe('c-tag', () => {
  it('renders default html', async () => {
    const page = await newSpecPage({
      components: [CTag],
      html: `<c-tag></c-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders active tag', async () => {
    const page = await newSpecPage({
      components: [CTag],
      html: `
      <c-tag active=${true}></c-tag>

      <c-tag active=${true}></c-tag>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('make tag active on click', async () => {
    const page = await newSpecPage({
      components: [CTag],
      html: `
      <c-tag active=${false}></c-tag>`,
    });
    expect(page.root).toMatchSnapshot();

    const tagToggle = page.root.shadowRoot.querySelector(
      '.c-tag',
    ) as HTMLElement;

    tagToggle.click();

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });
});
