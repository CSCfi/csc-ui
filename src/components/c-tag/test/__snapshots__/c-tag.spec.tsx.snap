// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-tag make tag active on click 1`] = `
<c-tag active="false" role="button" tabindex="0">
  <mock:shadow-root>
    <div class="c-tag">
      <div class="row">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</c-tag>
`;

exports[`c-tag make tag active on click 2`] = `
<c-tag active="false" role="button" tabindex="0">
  <mock:shadow-root>
    <div class="c-tag">
      <div class="row">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</c-tag>
`;

exports[`c-tag renders active tag 1`] = `
<c-tag active="true" role="button" tabindex="0">
  <mock:shadow-root>
    <div class="active c-tag">
      <div class="row">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</c-tag>
`;

exports[`c-tag renders default html 1`] = `
<c-tag role="button" tabindex="0">
  <mock:shadow-root>
    <div class="c-tag">
      <div class="row">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
</c-tag>
`;
